greedy_algorithms/boruvkas_minumum_spanning_tree.cpp

```diff
@@ -0,0 +1,217 @@
+			for (int j = i+1; j < size; j++) {
+
+				if (adj[i][j] == INT_MAX || adj[i][j] == 0) {  // No connection
+                    continue;
+                }
+
+				//Finds the parents of the start and end points to make sure they arent in the same group
+				int parentA = findParent(parent, i);
+				int parentB = findParent(parent, j);
+
+				if (parentA != parentB) {
+
+					//Grabs the start and end points for the first groups current smallest edge
+					int start = smallest_edge[parentA].first;
+					int end = smallest_edge[parentA].second;
+
+					//If there is no current smallest edge, or the new edge is smaller, records the new smallest
+					if (start == -1 || adj [i][j] < adj[start][end]) {
+						smallest_edge[parentA].first = i;
+						smallest_edge[parentA].second = j;
+					}
+
+					//Does the same for the second group
+					start = smallest_edge[parentB].first;
+					end = smallest_edge[parentB].second;
+
+					if (start == -1 || adj[j][i] < adj[start][end]) {
+						smallest_edge[parentB].first = j;
+						smallest_edge[parentB].second = i;
+					}
+				}
+			}
+		}
+
+		//Step 3: Combine the groups based off their smallest edge
+
+		for (int i = 0; i < size; i++) {
+						
+			//Makes sure the smallest edge exists
+			if (smallest_edge[i].first != -1) {
+
+				//Start and end points for the groups smallest edge
+				int start = smallest_edge[i].first;
+				int end = smallest_edge[i].second;
+
+				//Parents of the two groups - A is always itself
+				int parentA = i;
+				int parentB = findParent(parent, end);
+
+				//Makes sure the two nodes dont share the same parent. Would happen if the two groups have been 
+				//	merged previously through a common shortest edge
+                if (parentA == parentB) {
+                    continue;
+                }
+
+				//Tries to balance the trees as much as possible as they are merged. The parent of the shallower
+				//	tree will be pointed to the parent of the deeper tree.
+				if (parent[parentA].second < parent[parentB].second) {
+					parent[parentB].first = parentA; //New parent
+					parent[parentB].second++; //Increase depth
+				}
+				else {
+					parent[parentA].first = parentB;
+					parent[parentA].second++;
+				}
+				//Add the connection to the MST, using both halves of the adj matrix
+				MST[start][end] = adj[start][end];
+				MST[end][start] = adj[end][start];
+				total_groups--;	//one fewer group
+			}
+		}
+	}	
+	return MST;	
+}
+} // namespace greedy_algorithms
+
+/**
+ * @brief counts the sum of edges in the given tree
+ * @param 2d vector adjacency matrix
+ * @returns the int size of the tree
+ */
+int test_findGraphSum(std::vector<std::vector<int>> adj) {
+
+	int size = adj.size();
+	int sum = 0;
+
+	//Moves through one side of the adj matrix, counting the sums of each edge
+	for (int i = 0; i < size; i++) {
+		for (int j = i + 1; j < size; j++) {
+			if (adj[i][j] < INT_MAX) {
+				sum += adj[i][j];
+			}
+		}
+	}
+	return sum;
+}
```

Comment:
```suggestion
}
}  // namespace greedy_algorithms
```


__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix